Revert "Make nodepool concurrent ops scale better (#12488)"

Xylosma opened the pull request:
This reverts commit 1dbed42bb9ea904991e432c09a28fffca929452e.

<!--
Complete the self-review checklist to help speed up the review process: https://googlecloudplatform.github.io/magic-modules/contribute/review-pr/

If your PR is still work in progress, please create it in draft mode.

Put a description of what this PR is for here, along with any references to issues that this resolves or contributes to.
For example: Fixes https://github.com/hashicorp/terraform-provider-google/issues/ISSUE_ID
-->

**Release Note Template for Downstream PRs (will be copied)**
```release-note:bug
container: reverted locking behavior that caused regression of operation apply time spike started in `v6.15`
```

KatrinaHoffert reviewed (approved):
LGTM from GKE's side.

We might wanna tweak the proposed release note, though? Mark it as type bug (https://googlecloudplatform.github.io/magic-modules/code-review/release-notes/#type-specific-guidelines-and-examples). Probably don't use the word "commit" and mention when the issue started (v6.15). There doesn't seem to be a lot of past precedent for how to word it, but I found one similar case that we may want to copy from:

> vpcaccess: reverted new behaviour introduced by resource `google_vpc_access_connector` in `4.75.0`. `min_throughput` and `max_throughput` fields lost their default value, and customers could not make deployment due to that change.

Xylosma commented:
> LGTM from GKE's side.
> 
> We might wanna tweak the proposed release note, though? Mark it as type bug (https://googlecloudplatform.github.io/magic-modules/code-review/release-notes/#type-specific-guidelines-and-examples). Probably don't use the word "commit" and mention when the issue started (v6.15). There doesn't seem to be a lot of past precedent for how to word it, but I found one similar case that we may want to copy from:
> 
> > vpcaccess: reverted new behaviour introduced by resource `google_vpc_access_connector` in `4.75.0`. `min_throughput` and `max_throughput` fields lost their default value, and customers could not make deployment due to that change.

Added the impacted version.
The release note was already a type bug, was there anywhere else that I should mark it?

roaks3 reviewed (commented):
For reference, here was the original PR https://github.com/GoogleCloudPlatform/magic-modules/pull/12488

@Xylosma could you just confirm that other than the apply time, there is no other impact here? I want to be sure this isn't a breaking change, since the previous change is already out in the wild.

modular-magician commented:
Hi there, I'm the Modular magician. I've detected the following information about your changes:

## Diff report

Your PR generated some diffs in downstreams - here they are.

`google` provider: [Diff](https://github.com/modular-magician/terraform-provider-google/compare/auto-pr-12916-old..auto-pr-12916) ( 2 files changed, 175 insertions(+), 147 deletions(-))
`google-beta` provider: [Diff](https://github.com/modular-magician/terraform-provider-google-beta/compare/auto-pr-12916-old..auto-pr-12916) ( 2 files changed, 175 insertions(+), 147 deletions(-))

modular-magician commented:
#### Tests analytics
Total tests: 225
Passed tests: 214
Skipped tests: 11
Affected tests: 0

<details>
<summary>Click here to see the affected service packages</summary>
<blockquote>

<ul>
<li>container</li>

</ul>

</blockquote>
</details>

🟢 All tests passed!

View the [build log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-12916/artifacts/83bb136f-f96b-4bd6-a6af-39e3f8d5b9b8/build-log/replaying_test.log)

KatrinaHoffert commented:
This should not be a breaking change, at least not in the typical sense. The previous PR unintentionally mostly removed parallelism, since the cluster write lock would never be acquireable when we're waiting for a node pool to be created/updated/deleted (except for a brief race condition when a node pool request just finished but we haven't yet acquired the read lock while it waits for the operation). This restores parallelism as it previously worked.

If anything, the previous PR could be viewed as somewhat breaking because the lack of parallelism could cause TF apply to timeout due to taking longer. For any users impacted in that way, temporarily reducing parallelism until this revert is live should fix that. That works because the timeouts are counting down even when waiting for these locks. Such a workaround would only be for fixing timeouts caused by this issue. This PR is required to fix the apply time.

roaks3 reviewed (approved):
Thanks, that makes sense!
